sdk/core/azure-core/test/ut/bodystream.cpp

```diff
@@ -39,26 +39,40 @@ TEST(BodyStream, Rewind)
 
 #if defined(AZ_PLATFORM_POSIX)
   testDataPath.append("/fileData");
-  int f = open(testDataPath.data(), O_RDONLY);
-  EXPECT_GE(f, 0);
 #elif defined(AZ_PLATFORM_WINDOWS)
   testDataPath.append("\\fileData");
-  HANDLE f = CreateFile(
-      testDataPath.data(),
-      GENERIC_READ,
-      FILE_SHARE_READ,
-      NULL,
-      OPEN_EXISTING,
-      FILE_FLAG_SEQUENTIAL_SCAN,
-      NULL);
-  EXPECT_NE(f, INVALID_HANDLE_VALUE);
 #else
 #error "Unknown platform"
 #endif
-  auto fileBodyStream = Azure::Core::IO::FileBodyStream(f, 0, 0);
+
+  Azure::Core::IO::FileBodyStream fileBodyStream(testDataPath);
   EXPECT_NO_THROW(fileBodyStream.Rewind());
 
   std::vector<uint8_t> data = {1, 2, 3, 4};
   Azure::Core::IO::MemoryBodyStream ms(data);
   EXPECT_NO_THROW(ms.Rewind());
 }
+
+TEST(FileBodyStream, BadInput)
+{
+  EXPECT_THROW(Azure::Core::IO::FileBodyStream(""), std::runtime_error);
+  EXPECT_THROW(Azure::Core::IO::FileBodyStream("FileNotFound"), std::runtime_error);
+}
+
+constexpr int64_t FileSize = 1024 * 100;
+
+TEST(FileBodyStream, Length)
+{
+  std::string testDataPath(AZURE_TEST_DATA_PATH);
+  testDataPath.append("/fileData");
+
+  Azure::Core::IO::FileBodyStream stream(testDataPath);
+  EXPECT_EQ(stream.Length(), FileSize);
+
+  auto readResult = Azure::Core::IO::BodyStream::ReadToEnd(
+      stream, Azure::Core::Context::GetApplicationContext());
+  EXPECT_EQ(readResult.size(), FileSize);
```

Comment:
Do we want to validate the contents of the Read as well?  I think we can also use EXPECT_EQ
EXPECT_EQ(std::string(readResult.begin(), readResult.end()), "expected string")

Reply:
This is just testing the length, we have existing tests that validate the read. The file is also relatively large, 100kb of just `X`, so it isn't worth comparing to an expected string.